src/main/java/org/jabref/gui/groups/GroupTreeController.java

```diff
@@ -81,10 +81,10 @@ public void initialize() {
                 (newSelectedGroups) -> newSelectedGroups.forEach(this::selectNode);
         Consumer<List<TreeItem<GroupNodeViewModel>>> updateViewModel =
                 (newSelectedGroups) -> {
-                    if (newSelectedGroups == null) {
-                        viewModel.selectedGroupsProperty().clear();
-                    } else {
+                    try {
                         viewModel.selectedGroupsProperty().setAll(newSelectedGroups.stream().map(TreeItem::getValue).collect(Collectors.toList()));
+                    } catch (NullPointerException e) {
```

Comment:
Why do you catch a NPE here? Looks strange

Reply:
I know. For some reason this Binding triggers a NullPointerException when a group is removed, even though there was a null check directly before. I tried hard, but I could not pin down the actual cause of this (or which object actually is null). The debugger is really of no use with lambdas and bindings. But look at the complete code change in this class, it's just a different way of doing the same thing with a null check.

Apart from that, the NPE doesn't have any effect. The removal works anyway and the selection is cleared. All that this line does is to avoid a stack trace in the log.

I can offer to add a code comment explaining this, but I don't have more time to investigate why this is triggered (spent enough on that already). So it's either this solution or a stack trace in the log.

Reply:
Okay, add a comment and I am fine. Not the best idea yeah but I know how hard it is do debug binding lamdba stuff

Reply:
Okay, I took the time to rewrite that part in an imperative fashion, track down the bug, and fix it properly. It is totally insane... `List<TreeItem<GroupNodeViewModel>> newSelectedGroups` is a list that can actually contain `null` values. When you iterate over that list in an absolutely standard fashion, e.g. `for (TreeItem<GroupNodeViewModel> model : newSelectedGroups) {` than `model` can actually be null.

If that's not a violation of the basic contract of how collections work in Java, then I don't know what would be.

Reply:
This is really odd

Reply:
The NPE is valid in the for each loop, it does not check against NULL, it only skips the loop if the list is empty. So this is more of a design issue that the TreeItem could be null at one poin
https://stackoverflow.com/questions/85190/how-does-the-java-for-each-loop-work?rq=1

Reply:
Interesting, thanks :) You never finish learning about even the most basic aspects of the language.

Anyway, the code avoids the NPE now through proper checking, so this PR should be ready to go.